File: PathBot.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class PathBot {
    public static void main(String[] args) {
        Map<String, Object> initialPosition = new HashMap<>();
        initialPosition.put("axis", new int[]{0,0});
        initialPosition.put("direction", "NORTH");

        String[] LEFT = new String[]{"L", "WEST"};
        String[] RIGHT = new String[]{"R", "EAST"};
        String[] NORTH = new String[]{"N", "NORTH"};
        String[] SOUTH = new String[]{"L", "SOUTH"};


        for (String arg : args) {
            String direction = String.valueOf(arg.charAt(0));
            int moveN = Integer.parseInt(arg.substring(2));
            int[] currentAxis = (int[]) initialPosition.get("axis");
            String currentDirection = (String) initialPosition.get("direction");

            if (RIGHT[0].equalsIgnoreCase(direction)) {
                initialPosition.put("axis", new int[]{currentAxis[0] + moveN, currentAxis[1]});
                switch (currentDirection) {
                    case "NORTH":
                        initialPosition.put("direction", RIGHT[1]);
                        break;
                    case "EAST":
                        initialPosition.put("direction", SOUTH[1]);
                        break;
                    case "SOUTH":
                        initialPosition.put("direction", LEFT[1]);
                        break;
                    case "WEST":
                        initialPosition.put("direction", NORTH[1]);
                        break;
                }
            }
            if (LEFT[0].equalsIgnoreCase(direction)) {
                initialPosition.put("axis", new int[]{currentAxis[0] - moveN, currentAxis[1]});
                switch (currentDirection) {
                    case "NORTH":
                        initialPosition.put("direction", LEFT[1]);
                        break;
                    case "WEST":
                        initialPosition.put("direction", SOUTH[1]);
                        break;
                    case "SOUTH":
                        initialPosition.put("direction", RIGHT[1]);
                        break;
                    case "EAST":
                        initialPosition.put("direction", NORTH[1]);
                        break;
                }
            }
        }

        int[] finalPosition = (int[]) initialPosition.get("axis");
        System.out.println(initialPosition.get("direction") + String.format("(%s,%s)", finalPosition[0], finalPosition[1]));
    }
}

File: path-bot.py
import sys

if len(sys.argv) < 2:
    print("Please enter path")
    sys.exit(1)

plaintext = sys.argv[1:]


"""

L -> (Left or West)
R -> (Right or East)
N -> (North )
S -> (South)
Mn -> Move ( M10 -> move 10 forward )

Eg.  
1. RM15  ==>  Right + move to 15   => East (15, 0)

2. RM15 LM10 
==>  (Right + move to 15 )   , (Left + move to 10)
==> East(15,0) + (Left + move to 10)
==>  
"""

initial_position = {"axis": (0,0), "direction": "NORTH"}
LEFT = ("L","WEST" )
RIGHT = ("R", "EAST")
NORTH = ("N", "NORTH")
SOUTH = ("S", "SOUTH")
for text in plaintext:
    direction = text[0]
    moveN = int(text[2:])
    if(direction in RIGHT):
        initial_position["axis"] = (initial_position["axis"][0] + moveN, initial_position["axis"][1])
        match initial_position["direction"]:
            case 'NORTH':
                initial_position["direction"] = RIGHT[1]
            case 'EAST':
                initial_position["direction"] = SOUTH[1]
            case 'SOUTH':
                initial_position["direction"] = LEFT[1]
            case 'WEST' :
                initial_position["direction"] = NORTH[1]

    if(direction in LEFT):
        initial_position["axis"] = (initial_position["axis"][0] - moveN, initial_position["axis"][1])
        match initial_position["direction"]:
            case 'NORTH':
                initial_position["direction"] = LEFT[1]
            case 'WEST':
                initial_position["direction"] = SOUTH[1]
            case 'SOUTH':
                initial_position["direction"] = RIGHT[1]
            case 'EAST' :
                initial_position["direction"] = NORTH[1]

    # break

print(initial_position["direction"]+str(initial_position["axis"]))
